docs/current/742989-cookbook.md

```diff
@@ -625,6 +625,44 @@ Set the Hashicorp Vault URI, namespace, role and secret identifiers as host envi
 
 [Learn more](./guides/723462-use-secrets.md)
 
+## Error handling
+
+### Aborting with unhandled error
+
+The following code listing demonstrates how to abort the script with an unhandled error, which allows debugging with the stack trace of the crash during development.
+
+```go file=./cookbook/snippets/error-handling/unhandled/main.go
+```
+
+### Aborting gracefully
+
+The following code listing demonstrates how to abort the script without crashing.
+
+```go file=./cookbook/snippets/error-handling/aborting/main.go
+```
+
+### Exit code and unexpected errors
+
+The following code listing demonstrates how to handle a non-zero exit code from running a command, with several use cases:
```

Comment:
I found this example quite difficult to grok because it covers multiple use cases. I recommend splitting this into a separate recipe for each use case for readability.

Reply:
Yes, OK to merge as is and split into separate snippets later. Added a note for this in https://github.com/dagger/dagger/issues/5539#issuecomment-1662082694